scripts/library.sh

```diff
@@ -615,19 +620,29 @@ function get_latest_knative_yaml_source() {
   local yaml_name="$2"
   # If it's a release branch, the yaml source URL should point to a specific version.
   if is_release_branch; then
-    # Get the latest tag name for the current branch, which is likely formatted as v0.5.0
-    local tag_name="$(git describe --tags --abbrev=0)"
-    # The given repo might not have this tag, so we need to find its latest release manifest with the same major&minor version.
-    local major_minor="$(echo ${tag_name} | cut -d. -f1-2)"
-    local yaml_source_path="$(gsutil ls gs://knative-releases/${repo_name}/previous/${major_minor}.*/${yaml_name}.yaml \
+    local major_minor="UNKNOWN"
+    # Extract the release major&minor version from either the latest tag or the branch name.
+    local tag_name="$(git describe --tags --abbrev=0 2> /dev/null)"
```

Comment:
`git describe --tags --abbrev=0` will return the tag name whose commit ID is nearest to the current branch (sorry the comment was not accurate). For instance, if `release-0.13` has been cut but not been tagged, this command will return `v0.12.0`. So this approach won't work.

Instead of using tags, how about we only check the branch name by only keeping line 629-630?

Reply:
Thanks, updated.